src/install-update.ts

```diff
@@ -0,0 +1,167 @@
+import { checkForSDKUpdates, Release } from "./check-update.ts";
+import { getJSON } from "./utilities.ts";
+import { projectScripts } from "./mod.ts";
+
+export const SDK_NAME = "the Slack SDK";
+
+interface InstallUpdateResponse {
+  name: string;
+  updates: Update[];
+  error?: {
+    message: string;
+  } | null;
+}
+
+export interface Update {
+  name: string;
+  current: string;
+  latest: string;
+  error?: {
+    message: string;
+  } | null;
+}
+
+/**
+ * updateDependencies checks for SDK-related dependency updates. If
+ * updatable releases are found, dependency files are updated with the
+ * latest dependency versions and the project changes are cached.
+ */
+export const updateDependencies = async () => {
+  const { releases } = await checkForSDKUpdates();
+  const updatableReleases = releases.filter((r) => r.current && r.latest);
+  const updateResp = await createUpdateResp(updatableReleases);
+
+  // If no errors occurred during installation, re-build
+  // project as a means to cache the changes
+  if (!updateResp.error) runBuildHook();
+
+  return updateResp;
+};
+
+/**
+ * createUpdateResp creates an object that contains each update,
+ * featuring information about the current and latest version, as well
+ * as if any errors occurred while attempting to update each.
+ */
+export async function createUpdateResp(
+  releases: Release[],
+): Promise<InstallUpdateResponse> {
+  const updateResp: InstallUpdateResponse = { name: SDK_NAME, updates: [] };
+
+  if (!releases.length) return updateResp;
+
+  try {
+    const cwd = Deno.cwd();
+
+    // Update import_map.json with latest dependency versions
+    const importsUpdateResp = await updateDependencyFile(
+      `${cwd}/import_map.json`,
+      releases,
+    );
+
+    // Update slack.json with latest dependency versions
+    const hooksUpdateResp = await updateDependencyFile(
+      `${cwd}/slack.json`,
+      releases,
+    );
+
+    updateResp.updates = [...importsUpdateResp, ...hooksUpdateResp];
+  } catch (err) {
+    updateResp.error = { message: err.message };
+  }
+
+  return updateResp;
+}
+
+/**
+ * updateDependencyFile reads the contents of a provided path, calls updateDependencyMap
+ * to swap out the current versions with the latest releases available, and then writes
+ * to the dependency file. Returns a summary of updates made.
+ */
+export async function updateDependencyFile(
+  path: string,
+  releases: Release[],
+): Promise<Update[]> {
+  try {
+    const dependencyMap = await getJSON(path);
+    const { imports, hooks } = dependencyMap;
+
+    // If file doesn't exist or expected dependency key is missing
+    if (!imports && !hooks) return [];
```

Comment:
So if the file doesn't exist, based on the current behaviour of `getJSON`, we won't know about that, since we ignore any errors raised from that method. I think we should change that behaviour so that `getJSON` will raise an error, and thus the `catch` block in here would catch the missing-file error, and raise that to the user. I think that is valuable context to provide to the user for this particular failure mode.

Reply:
I know I responded and indicated I was going to punt on this, and still might, but just for clarity: 

- If the user is missing a `slack.json` file, do we want to mimic the same response that is currently raised on the CLI-side of things (ie, `Error: Current directory is not a Slack project`)?

- If we cannot find an `import_map.json` file, and we are not acknowledging (yet) other valid dependency files, what language do you propose we raise?

Reply:
The granularity of which method returns which error matters in this situation, I think. `getJSON` might error for a variety of reasons beyond just a missing file: maybe the permissions on the file changed, or maybe someone (maybe even our own tools from a previous update!) edited the file and caused it to become invalid JSON. With that in mind, re: your questions:

> If the user is missing a slack.json file, do we want to mimic the same response that is currently raised on the CLI-side of things (ie, Error: Current directory is not a Slack project)?

If we're positive that the error is a file-not-found error, then I think your suggestion makes sense. But since there are other kinds of errors that could lead to being unable to read slack.json, perhaps a more generic one is applicable? Something like "Could not read/update slack.json: <actual system error returned>". This generic-enough message may be good enough for both slack.json and import_map.json, too.